docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,76 @@
+---
+title: 'Documentation Versioning'
+linkTitle: 'Documentation Versioning'
+---
+
+- Author: [Uchechukwu Obasi](https://github.com/thisisobate)
+- Date: May 2020
+- Status: proposal
+
+## Problem
+
+The Thanos website contains the docs area which is built by rendering the markdown files from the tip of master of the Thanos repository. This frequently causes confusion for users, as breaking changes are often merged into master that conflicts with the APIs of previous releases.
+
+In this proposal, we want to solve this. In particular, we want to:
+
+1. Build a Version Picker that serves as a tool which will aid easy access to other versions of the documentation. This picker will Include links to change version (the version must be in the URL).
+2. A solid and robust Documentation structure.
+3. Version specific pages of the documentation
+4. Design a workflow for managing docs that integrates with Thanos Git workflow ie updating corresponding docs on pull requests, cherry-picks, etc.
+
+## Proposed Solution
+
+### Version Picker
+
+Currently, the documentation is residing under the docs/ folder of Thanos repository. It is built by Hugo. It will have a proper drop-down menu just like prometheus [`drop-down menu`](https://prometheus.io/docs/introduction/overview/) which will enable proper versioning. This user facing tool will be built using HTML and CSS.
+
+### Documentation Structure
+
+During the course of my research, I was able to deduce three different approaches of which we can solve this documentation structure issue.
+
+1. Sub branching: Creating different sub branches in the content folder of the Thanos repository. Since the current architecture of Hugo is this:
+
+|- archetypes
+|- config.toml
+|- content
+|- data
+|- layouts
+|- static
+|- themes
+
+You can add aditional versioned content in different folders within the content directory. For example:
+
+content
+|- docs
+|- master
+|- docs-v2
+|- docs-v3
+|- other-folder-for-other-releases
+
+_Note: The current version lives in the `docs` folder_
+
+#### Example:
+
+For `master` version we would be able to use the above links via the following path
+
+```
+/docs/master/components/
+```
+
+And for a minor version like `1.x`:
+
+```
+/docs/docs-v2/components/
+```
+
+2. Version Branching: You can decide to make each individual version to be a separate branch of it's own in the master repository.
```

Comment:
We already do that. Each our minor release has it's own branch. `release-0.X` we can leverage that :hugs:

Reply:
Wow! Awesome! I believe that will make the work easier for us!